packages/ERTP/src/issuer.js

```diff
@@ -154,75 +158,95 @@ function produceIssuer(allegedName, mathHelpersName = 'nat') {
       purseLedger.init(purse, amountMath.getEmpty());
       return purse;
     },
-    isLive: paymentLedger.has,
-    getAmountOf: paymentLedger.get,
+
+    isLive: paymentP => {
+      return Promise.resolve(paymentP).then(payment => {
+        return paymentLedger.has(payment);
+      });
+    },
+    getAmountOf: paymentP => {
+      return Promise.resolve(paymentP).then(payment => {
+        let result;
+        try {
```

Comment:
Don't use a `try/catch` under the assumption that you know why the caught error was thrown. If something else was broken, this will mask the diagnostic error. Instead, start with
```js
assert(paymentLedger.has(payment), details`...`);
```

Later, we should check all `try/catch`es in our code to see if they're making a similar bad assumption.

Reply:
done